test/version_solver_test.dart

```diff
@@ -284,6 +284,130 @@ void devDependency() {
     await d.appDir({'foo': '1.0.0'}).create();
     await expectResolves(result: {'foo': '1.0.0'});
   });
+
+  group("with both a dev and regular dependency", () {
+    test("succeeds when both are satisfied", () async {
+      await servePackages((builder) {
+        builder.serve('foo', '2.0.0');
```

Comment:
Might be good to serve 1.0.0 and 3.0.0 to ensure that the chosen package is also satisfies both ranges.